homeassistant/loader.py

```diff
@@ -979,6 +980,39 @@ def get_platform(self, platform_name: str) -> ModuleType:
             return platform
         return self._load_platform(platform_name)
 
+    def get_integration_platform(self, platform_name: str) -> ModuleType:
```

Comment:
Maybe the name isn't so great here

Reply:
Why make this a new method and not put it in the normal load platform method ?

Reply:
The likely outcome here is a failure to load where-as the normal load platform has a likely outcome of success

Reply:
I've redesigned this